### PR TITLE
Add mockComponent

### PR DESCRIPTION
**Pull request by oliverviljamaa**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,20 +28,34 @@
...
     "babel-core": "^6.26.3",
+    "babel-plugin-transform-object-rest-spread": "^6.26.0",
     "babel-preset-env": "^1.7.0",
     "eslint": "^5.2.0",
     "eslint-config-airbnb-base": "^13.0.0",
     "eslint-config-prettier": "^2.9.0",
     "eslint-plugin-import": "^2.13.0",
     "eslint-plugin-prettier": "^2.6.2",
     "husky": "^0.14.3",
+    "jest": "^23.4.2",
```
Comment (codewhisperer) [inline package.json]: This component is probs not backwards compatible with Jasmine anymore is it?

Comment (oliverviljamaa) [reply package.json]: It is, it doesn't make any assumptions on the test framework/runner, I'll update the README, thanks!

File: src/mockComponent/mockComponent.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  };
+  return mock;
+}
+
+function withProp(mock) {
+  mock.prop = key => mock._controller[key];
+  return mock;
+}
+
+function withSimulate(mock) {
```
Comment (codewhisperer) [inline src/mockComponent/mockComponent.js]: 😍

File: src/mockComponent/mockComponent.spec.js [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+            some-prop="someProp"
+            on-some-prop-change="onSomePropChange()"
+          ></some-component>
+        `,
+        props,
+      );
+
+      expect(mockedComponent.props()).toEqual({
+        someProp: 'Some value',
+        onSomePropChange: expect.any(Function), // due to Angular's handling of callbacks
```
Comment (codewhisperer) [inline src/mockComponent/mockComponent.spec.js]: Nice!

**Pull request merged**